### PR TITLE
Topology Class

### PR DESCRIPTION
**Pull request by ziadsawalha**
Use ExtensibleDict to implement a Topology class. 

Ported from #21.

This PR should be used to review e3de04a only.

### CODE REVIEW

File: chessboard/topology.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+    """Construct and validate the relations between application services.
+
+    The main thing this class does right now is read parsed and validated
+    Checkmatefile contents (see :meth:`from_deployment`), analyze the
+    relations defined for each service, create a simple mapping for those
+    relations, and check that the relations are valid.
+    """
+
+    X__schema__ = schema.BLUEPRINT_SCHEMA
```
Comment (larsbutler) [inline chessboard/topology.py]: This is a funny variable name. Is this a typo?

File: tox.ini [hunk 2]
```diff
@@ -8,8 +8,7 @@ envlist = py27,py34,style,docs
 install_command = pip install -U {opts} {packages}
 deps = -r{toxinidir}/requirements.txt
        -r{toxinidir}/test-requirements.txt
-commands = nosetests {posargs} --with-doctest --with-coverage --cover-package=chessboard \
-    --with-ignore-docstrings
+commands = nosetests {posargs} --with-doctest --with-coverage --cover-package=chessboard
```
Comment (larsbutler) [inline tox.ini]: Why did you remove `--with-ignore-docstrings`? This pretty much makes the verbose output of nosetests useless. :(

Comment (larsbutler) [inline tox.ini]: Doh! This already got merged. Reverted in https://github.com/checkmate/chessboard/pull/34.

File: chessboard/topology.py (hunk 1 above)
Comment (ziadsawalha) [inline chessboard/topology.py]: Ah yes. I was having trouble tracking a schema bug and that  was a quick way for me to disable enforcing it on this class. Good catch.

File: tox.ini (hunk 2 above)
Comment (ziadsawalha) [inline tox.ini]: Hold on - it's not merged yet... I can remove it.

Comment (ziadsawalha) [inline tox.ini]: Oh, I see. A rebase will show it merged. Got it now..

**Comment by larsbutler** [comment]
Looks like this branch needs a rebase.

**Comment by larsbutler** [comment]
LGTM
